File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
// const input = fs.readFileSync('./input.txt').toString().trim().split('\n');
const N = +input[0];
const arr = input.slice(1);

const solution = (N, arr) => {
  const mp = new Map();
  let ret = 0;


  arr.forEach((e, i) => {
    if(e === "ENTER"){
      mp.clear();
    }else {
      if(!mp.has(e)){
        mp.set(e, 1);
        ret++;
      }
    }
  })
  
  return ret;
}

console.log(solution(N, arr));

File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.cc
#include <bits/stdc++.h>
using namespace std;

int N, ret;
string s;
map<string, int> mp;

int main() {
  cin >> N;

  for (int i = 0; i < N; i++) {
    cin >> s;
    if (s == "ENTER") {
      mp.clear();
    } else {
      mp[s]++;
      if (mp[s] == 1) {
        ret++;
      }
    }
  }

  cout << ret << '\n';
  return 0;
}